### PR TITLE
Handle the start event

### PR DESCRIPTION
**Pull request by mthaddon**
We need to set `ActiveStatus` in the `start` event because this is a workload-less charm, so there's no `pebble-ready` event to react to. This means this is the only event (outside of `update-status`) that is fired if a pod is restarted by the k8s cluster. If we don't do anything here the charm would remain in maintenance status.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -518,6 +519,19 @@ def _on_config_changed_with_warning(self, event: Any) -> None:
             )
             self.unit.status = status.from_name(status.name, warning)
 
+    def _on_start(self, event: Any) -> None:
```
Comment (arturo-seijas) [inline src/charm.py:522]: It'd probably be better to use an underscore instead of `event` and don't document the arg in the docstring

Comment (mthaddon) [reply src/charm.py:522]: Good point, updated.

File: src/charm.py [hunk 2]
```diff
@@ -518,6 +519,15 @@ def _on_config_changed_with_warning(self, event: Any) -> None:
             )
             self.unit.status = status.from_name(status.name, warning)
 
+    def _on_start(self, _: Any) -> None:
```
Comment (arturo-seijas) [inline src/charm.py:522]: S/Any/StartEvent

Comment (mthaddon) [reply src/charm.py]: Have updated.

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by weiiwang01 — APPROVED** [review]

**Pull request merged**